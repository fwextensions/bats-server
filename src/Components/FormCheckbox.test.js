import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import FormCheckbox from './FormCheckbox';

test('Snapshot', () => {
  const component = renderer.create(
    <FormCheckbox
      currentValue="apple"
      label="banana"
      onChange={() => {}}
      property="banana"
      value={true}
    />);

    const componentJson = component.toJSON();
    expect(componentJson).toMatchSnapshot();
});

test('FormInput should trigger the onChange callback', () => {
  const mock = jest.fn();

  const wrapper = render(
    <FormCheckbox
      currentValue="apple"
      label="Coconut"
      onChange={mock}
      property="coconut"
      value={true}
    />);

    const checkbox = wrapper.getByLabelText('Coconut');
    fireEvent.click(checkbox);
    expect(mock.mock.calls.length).toBe(1);
})