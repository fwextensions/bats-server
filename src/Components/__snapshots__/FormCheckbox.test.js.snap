// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<div
  className="usa-checkbox"
>
  <input
    checked={false}
    className="usa-checkbox__input"
    disabled={false}
    id="banana-true"
    name="banana"
    onChange={[Function]}
    type="checkbox"
    value={true}
  />
  <label
    className="usa-checkbox__label"
    htmlFor="banana-true"
  >
    banana
  </label>
</div>
`;
